node/external/nodeApiResolver.go

```diff
@@ -131,6 +132,10 @@ func (nar *nodeApiResolver) ComputeTransactionGasLimit(tx *transaction.Transacti
 	return nar.txCostHandler.ComputeTransactionGasLimit(tx)
 }
 
+func (nar *nodeApiResolver) SimulateTransactionExecution(tx *transaction.Transaction) (*txSimData.SimulationResults, error) {
```

Comment:
missing comment, also, why do we have a new function for the tx simulation? I remember we already have 3 of them. Maybe clean them up?

Reply:
added comment